server/ctrlsubsonic/spec/construct_by_folder.go

```diff
@@ -101,6 +102,9 @@ func NewTCTrackByFolder(t *db.Track, parent *db.Album) *TrackChild {
 		trCh.Genres = append(trCh.Genres, &GenreRef{Name: g.Name})
 	}
 	for _, a := range t.Artists {
+		if a.Name == t.TagTrackArtist {
```

Comment:
what's this check for?

Reply:
Each `TrackChild` has a single `ArtistID`, which corresponds to the `TagTrackArtist`.  The check just makes sure that we assign the correct `ArtistID` to to the `TrackChild`. If this check wasn't here, it would assign the track's `ArtistID` to the last artist in the `Artists` list